at_secondary/at_secondary_server/test/update_verb_test.dart

```diff
@@ -678,27 +681,69 @@ void main() {
       var localLookUpResponse = Response();
       var localLookupVerbHandler = LocalLookupVerbHandler(keyStore);
       var localLookVerbParam = HashMap<String, String>();
-      localLookVerbParam.putIfAbsent(AT_SIGN, () => '@sitaram');
+      localLookVerbParam.putIfAbsent(AT_SIGN, () => '@alice');
       localLookVerbParam.putIfAbsent(AT_KEY, () => 'location');
       await localLookupVerbHandler.processVerb(
           localLookUpResponse, localLookVerbParam, atConnection);
       expect(localLookUpResponse.data, null);
       //Reset TTB
       updateVerbParams = HashMap<String, String>();
-      updateVerbParams.putIfAbsent(AT_SIGN, () => '@sitaram');
+      updateVerbParams.putIfAbsent(AT_SIGN, () => '@alice');
       updateVerbParams.putIfAbsent(AT_KEY, () => 'location');
       updateVerbParams.putIfAbsent(AT_TTB, () => '0');
       updateVerbParams.putIfAbsent(AT_VALUE, () => 'hyderabad');
       await updateVerbHandler.processVerb(
           updateResponse, updateVerbParams, atConnection);
       //LLOOKUP Verb - After TTB
-      localLookVerbParam.putIfAbsent(AT_SIGN, () => '@sitaram');
+      localLookVerbParam.putIfAbsent(AT_SIGN, () => '@alice');
       localLookVerbParam.putIfAbsent(AT_KEY, () => 'location');
       await localLookupVerbHandler.processVerb(
           localLookUpResponse, localLookVerbParam, atConnection);
       expect(localLookUpResponse.data, 'hyderabad');
     });
   });
+  group('A group of tests to validate sharedBy atsign', () {
+    test('sharedBy atsign is not equal to current atsign', () async {
+      var command = 'update:phone@bob +12345';
+      command = SecondaryUtil.convertCommand(command);
+      AbstractVerbHandler handler = UpdateVerbHandler(null);
+      AtSecondaryServerImpl.getInstance().currentAtSign = '@alice';
+      var secondaryPersistenceStore =
+          SecondaryPersistenceStoreFactory.getInstance()
+              .getSecondaryPersistenceStore(
+                  AtSecondaryServerImpl.getInstance().currentAtSign)!;
+      handler.keyStore = secondaryPersistenceStore
+          .getSecondaryKeyStoreManager()!
+          .getKeyStore();
+      var response = Response();
+      var verbParams = handler.parse(command);
+      var atConnection = InboundConnectionImpl(null, null);
+      expect(
+          () => handler.processVerb(response, verbParams, atConnection),
```

Comment:
processVerb is async ... does this `expect` definitely execute fully or should it be `await handler.processVerb` ?

Reply:
test was failing when I added async/await. Fixed the test now. expectLater is right method to use